packages/grafana-ui/src/components/Layout/Grid/Grid.internal.story.tsx

```diff
@@ -0,0 +1,342 @@
+import { Meta, StoryFn } from '@storybook/react';
+import React from 'react';
+
+import { GrafanaTheme2 } from '@grafana/data';
+
+import { useTheme2 } from '../../../themes';
+import { Stack } from '../../../unstable';
+import { withCenteredStory } from '../../../utils/storybook/withCenteredStory';
+
+import { Grid } from './Grid';
+import mdx from './Grid.mdx';
+import { GridItem } from './GridItem';
+
+const gridItem = (index: number, theme: GrafanaTheme2, width?: string, height?: string) => {
+  const colorOrder = index >= 6 ? index - 6 : index;
+  const itemOrder = index + 1;
+  const colorList = ['red', 'orange', 'purple', 'blue', 'yellow', 'green'].map(theme.visualization.getColorByName);
+  return (
+    <div
+      key={index}
+      style={{
+        backgroundColor: colorList[colorOrder],
+        width,
+        height,
+        fontSize: theme.typography.bodySmall.fontSize,
+        color: theme.colors.text.maxContrast,
+      }}
+    >
+      Item {itemOrder}
+    </div>
+  );
+};
+
+const meta: Meta<typeof Grid> = {
+  title: 'General/Layout/Grid',
+  component: Grid,
+  decorators: [withCenteredStory],
+  parameters: {
+    docs: {
+      page: mdx,
+    },
+  },
+  args: {
+    display: 'grid',
+    gap: 1,
+  },
+};
+
+export const Basic: StoryFn<typeof Grid> = (args) => {
+  const theme = useTheme2();
+  return (
+    <Grid
+      gap={args.gap}
+      rowGap={args.rowGap}
+      columnGap={args.columnGap}
+      templateColumns={args.templateColumns}
+      templateRows={args.templateRows}
+      autoFlow={args.autoFlow}
+    >
+      {Array.from({ length: 9 }).map((_, i) => gridItem(i, theme))}
+    </Grid>
+  );
+};
+
+Basic.args = {
+  templateColumns: 'repeat(3, 1fr)',
+  gap: 2,
+};
+
+Basic.parameters = {
+  controls: {
+    exclude: ['display', 'alignItems', 'justifyItems', 'autoRows', 'columnStart', 'columnEnd', 'rowStart', 'rowEnd'],
+  },
+};
+
+export const AlignItemsExamples: StoryFn<typeof Grid> = (args) => {
```

Comment:
I understand that it is meant to show different examples for alignment but it looks off as soon as the width is a little smaller.

<img width="838" alt="Screenshot 2023-09-19 at 16 58 44" src="https://github.com/grafana/grafana/assets/48948963/003452a6-200d-4c3e-80ad-a80bf70c95ae">

Reply:
Yes, I was thinking about changing the name to `Align items stretch (reference)`. Will it be understandable? 🤔